flask_project/campaign_manager/views.py

```diff
@@ -998,143 +907,160 @@ def create_campaign():
             map_type=data['map_type']
             )
         created_campaign.create()
-        save_campaign_feature_types(
-            created_campaign,
-            data
-            )
-        save_campaign_geometry(
-            created_campaign,
-            data
-            )
-        save_campaign_insight_functions(
-            created_campaign,
-            data
-            )
-        save_campaign_managers(
-            created_campaign,
-            data
-            )
+        created_campaign.save_feature_types(data)
+        created_campaign.save_geometry(data)
+        created_campaign.save_insight_functions(data)
+        created_campaign.save_managers(data)
         create_thumbnail_image(created_campaign)
         return redirect(
             url_for(
                 'campaign_manager.get_campaign',
                 uuid=data['uuid'])
         )
-    context = dict(
-        oauth_consumer_key=OAUTH_CONSUMER_KEY,
-        oauth_secret=OAUTH_SECRET,
-        map_provider=map_provider()
-    )
-    context['allowed_managers'] = managers
-    context['url'] = '/create'
-    context['action'] = 'create'
+
+
+def get_campaign_type(campaign_obj):
```

Comment:
this function could be a method of the object Campaign.
`campaign_obj.campaign_types`
that you can test.